### PR TITLE
Attempt to run without syscall tracepoints, if possible.

### PR DESCRIPTION
**Pull request by umanwizard**
This implements the proposal in
https://github.com/open-telemetry/opentelemetry-ebpf-profiler/issues/173 . Please see there for full details.

### CODE REVIEW

**Review by christos68k — COMMENTED** [review]
There are a few linter failutes. Other than that, since we now allow running the agent without syscall tracepoints, we should probably also remove the eBPF integration [test](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/e7a70381ba9c788280d5eaea5ae95628cc011e07/support/ebpf_integration_test.go#L1) and its dependent [hook](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/03072edfa5d1473774094d707687953c19b38da2/support/ebpf/interpreter_dispatcher.ebpf.c#L246-L255).

CC: @florianl

File: tracer/maccess.go [hunk 1]
```diff
@@ -26,7 +26,11 @@ func checkForMaccessPatch(coll *cebpf.CollectionSpec, maps map[string]*cebpf.Map
 
 	code, err := loadKernelCode(coll, maps, faultyFunc.Address)
 	if err != nil {
-		log.Warnf("Failed to load code for %s: %v", faultyFunc.Name, err)
+		log.Warnf("Failed to load code for %s: %v.\n"+
+			"Syscall tracepoints are not working on this system, so whether "+
+			"the kernel is patched could not be determined. "+
+			"Either use a kernel configured with syscall tracepoints, "+
+			"or upgrade to kernel version 6.4 or higher.", faultyFunc.Name, err)
```
Comment (florianl) [inline tracer/maccess.go:33]: `loadKernelCode` can fail for various reasons. Just naming tracepoints here might be misleading. 
```suggestion
		log.Warnf("Failed to load code for %s: %v\n"+
			"Upgrading the kernel version to 6.4 or higher could resolve the issue.", faultyFunc.Name, err)
```

**Review by florianl — COMMENTED** [review]
> [..] we should probably also remove the eBPF integration [test](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/e7a70381ba9c788280d5eaea5ae95628cc011e07/support/ebpf_integration_test.go#L1) and its dependent [hook](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/03072edfa5d1473774094d707687953c19b38da2/support/ebpf/interpreter_dispatcher.ebpf.c#L246-L255).

:+1: on that

File: tracer/maccess.go (hunk 1 above)
Comment (fabled) [reply tracer/maccess.go:33]: I think this warning/error should be moved from here to `loadKernelCode` or even `executeSystemAnalysisBpfCode` in the approariate place. Perhaps even checking the error code to see if its likely due to kernel config?

Comment (fabled) [reply tracer/maccess.go:33]: Is the flag `-no-kernel-version-check` to bypass this check mentioned in somewhere on this failure path?

**Comment by fabled** [comment]
@umanwizard Are you working on the feedback or do you need help to finalize this PR?

**Comment by umanwizard** [comment]
Sorry, I dropped this. I'll address the feedback and send a new version soon.

**Comment by umanwizard** [comment]
I'm not sure I understand why we should remove the integration test. Isn't it still useful to have this integration test run? We just need to ensure that it is always running in an environment that has syscall probes available (which is the vast majority of environments)

**Comment by florianl** [comment]
> I'm not sure I understand why we should remove the integration test.

With https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/205 a test was added that covers also the functionality of the named integration test for the tracepobe. And instead of a specific tracepoint that is written for the integration test, https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/205 tests the tracepoint that is actually used while running the eBPF profiler agent.

**Comment by umanwizard** [comment]
Okay. In that case we can indeed remove it.

**Review by florianl — COMMENTED** [review]
With the PR only the updated `support/ebpf/tracer.ebpf.release.arm64` is provided. Please also add the updated `support/ebpf/tracer.ebpf.release.amd64`.

**Review by florianl — APPROVED** [review]

**Review by fabled — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**